src/main/java/hudson/remoting/UserRequest.java

```diff
@@ -90,6 +90,17 @@ public UserRequest(Channel local, Callable<?,EXC> c) throws IOException {
         this.classLoaderProxy = RemoteClassLoader.export(cl, local);
     }
 
+    @Override
+    public void checkIfCanBeExecutedOnChannel(Channel channel) throws IOException {
+        // Default check for all requests
+        super.checkIfCanBeExecutedOnChannel(channel);
+        
+        // We also do not want to run UserRequests when the channel is being closed
+        if (channel.isClosingOrClosed()) {
+            throw new IOException("The request cannot be executed on channel " + channel + ". The channel is closing down or has closed down");
```

Comment:
Does this not just duplicate the check in `Channel.call` (as well as the check later in `Channel.send`)? I am unclear on why we are doing the same check three times in the course of one call.

(And again this should be `ChannelClosedException`.)